requirements/specifications/lifecycle/index.md

```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
 
-Unsuspending an app loads it's previous state back into memory and
-transitions it to the inactive state, so that it may be launched back
-into its original state.
+**TODO**: discuss ^^
 
-The platform may unsuspend apps for any number of reasons that are out
-of scope for this document.
+### 4.6. Resuming an app
+
+Resuming an app allows it to reallocate graphics composition and
+reload any resources it might have deallocated during `suspend()`.
+
+The platform may unsuspend apps so they can access more resources or
+in preperation to activate them.
 
 To unsuspend an app, platforms **MUST** use the following process.
 
-If an app is not in the suspended state, then it cannot be unsuspended
+If an app is not in the `SUSPENDED` state, then it cannot be unsuspended
 and there is no need to do anything else. The platform **MUST NOT**
 dispatch any *additional* lifecycle notifications when attempting to
-unsuspend an app that is not in the suspended state and the remainder of
-this section does not need to happen.
+unsuspend an app that is not in the `SUSPENDED` state and the remainder of
+this section does not apply.
+
+At this point, the app **MUST** be in the `SUSPENDED` state.
+
+Next, the platform **MUST** remove any restrictions on access the graphics
+compositor.
+
+Suspended apps **MUST** provide the  `xrn:firebolt:capability:lifecycle:application`
+capability, so the platform **MUST** call the app's implementation of
+`Application.resume()`:
+
+The platform **MUST** dispatch the `Lifecycle.onRequestResume`
+notification to the app, and wait for `appResumeTimeout` milliseconds
+for either a `Lifecycle.resumeResult` or `Lifecycle.resumeError`
+call in response.
+
+Once the platform receives the `resumeResult` call, then the app
+may be moved to the `RUNNING` state.
 
-At this point, the app **MUST** be in the suspended state.
+If the app times out or makes a `resumeError` call, then the app
+**MUST** be terminated.
 
-Next, the platform **MUST** load the app's saved memory state back into
-a browser container. If this fails for any reason then the unsuspend
-operation failed and should error out and the app **MUST** be considered
-unloaded. Additionally, the app **MUST** be removed from the list of
-suspended apps, so that an unsuspend operation is not attempted again.
+During the `resume()` transition, apps **SHOULD** reallocate graphics
+composition and other necessary resources.
 
-Finally, the platform **MUST** transition the app to the inactive state
-(see '[Inactive](#22-inactive)' above).
+### 4.7. Putting an app to sleep
+TBD
 
-### 3.7. Unloading an app
+Firebolt apps that have permission to use the `xrn:firebolt:capability:lifecycle:sleep` capability **MUST** implement `Sleepable.sleep()`.
+
+### 4.8. Waking an app from sleep
+TBD
+
+Firebolt apps that have permission to use the `xrn:firebolt:capability:lifecycle:sleep` capability **MUST** implement `Sleepable.wake()`.
+
+### 4.9. Unloading an app
```

Comment:
Andrew has pointed this out in a number of places but sounds like we should remove all mentions of "unloading" in favor of destroying.